test/DebugInfo/NonSemanticKernel100/DIModule.ll

```diff
@@ -0,0 +1,50 @@
+; ModuleID = '/Volumes/Data/apple-internal/llvm/tools/clang/test/Modules/debug-info-moduleimport.m'
+; RUN: llvm-as < %s -o %t.bc
+; RUN: llvm-spirv --spirv-debug-info-version=nonsemantic-kernel-100 %t.bc -o %t.spv
+; RUN: llvm-spirv -r -emit-opaque-pointers %t.spv -o - | llvm-dis -o %t.ll
+
+; RUN: llc -mtriple=x86_64-apple-macosx %t.ll -accel-tables=Dwarf -o %t -filetype=obj
+; RUN: llvm-dwarfdump -debug-info %t | FileCheck %s
+; RUN: llvm-dwarfdump -verify %t
+
+; RUN: llvm-spirv --spirv-debug-info-version=nonsemantic-kernel-100 %t.bc -spirv-text -o %t.spt
+; RUN: FileCheck %s --input-file %t.spt --check-prefix CHECK-SPIRV
+
+; CHECK: DW_TAG_compile_unit
+; CHECK-NOT: DW_TAG
+; CHECK:   DW_TAG_module
+; CHECK-NEXT: DW_AT_name {{.*}}"DebugModule"
+; CHECK-NEXT: DW_AT_LLVM_config_macros {{.*}}"-DMODULES=0"
+; CHECK-NEXT: DW_AT_LLVM_include_path {{.*}}"/llvm/tools/clang/test/Modules/Inputs"
+; CHECK-NEXT: DW_AT_LLVM_apinotes {{.*}}"m.apinotes"
+
+target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "spir64-unknown-unknown"
+
+; CHECK-SPIRV-DAG: ExtInstImport [[#EISId:]] "NonSemantic.Kernel.DebugInfo.100"
+; CHECK-SPIRV: String [[#FileName:]] "/llvm/tools/clang/test/Modules/<stdin>"
+; CHECK-SPIRV: String [[#EmptyStr:]] ""
+; CHECK-SPIRV: String [[#Name:]] "DebugModule"
+; CHECK-SPIRV: String [[#Defines:]] "-DMODULES=0"
+; CHECK-SPIRV: String [[#IncludePath:]] "/llvm/tools/clang/test/Modules/Inputs"
+; CHECK-SPIRV: String [[#ApiNotes:]] "m.apinotes"
+
+; CHECK-SPIRV: ExtInst [[#]] [[#Source:]] [[#]] DebugSource [[#FileName]]
+; CHECK-SPIRV: ExtInst [[#]] [[#Parent:]] [[#]] DebugCompileUnit 65536 4
+; CHECK-SPIRV: ExtInst [[#]] [[#SourceEmpty:]] [[#]] DebugSource [[#EmptyStr]]
+; CHECK-SPIRV: ExtInst [[#]] [[#Module:]] [[#]] DebugModule [[#Name]] [[#SourceEmpty]] 0 [[#Parent]] [[#Defines]] [[#IncludePath]] [[#ApiNotes]] 0
```

Comment:
```suggestion
; CHECK-SPIRV: ExtInst [[#]] [[#Module:]] [[#]] DebugModule [[#Name]] [[#SourceEmpty]] 0 [[#Parent]] [[#Defines]] [[#IncludePath]] [[#ApiNotes]] 0
```

One of the difference between OpenCL and NonSemantic debug info specs is that while OpenCL spec has literal parameters and NonSemantic doesn't have it. It means, that here instead of '0' should be result id of OpConstant. I'll created a draft PR that shows simple technic, that does this transformation for other instructions.

Reply:
I do not see any change here....

Reply:
It should be
; CHECK-SPIRV-DAG: OpConstant [[#TypeInt32]] [[#ConstNull:]] 0
.....
; CHECK-SPIRV: ExtInst [[#]] [[#Module:]] [[#]] DebugModule [[#Name]] [[#SourceEmpty]] **[[#ConstNull]]** [[#Parent]] [[#Defines]] [[#IncludePath]] [[#ApiNotes]] **[[#ConstNull]]**

And I can tell for sure that it's Literal by looking at https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1878/files#diff-ce5c478b99f08cea7d78a07829f628f520b4ab8746eb64c37874ad4f88fbccabR1263 and https://llvm.org/doxygen/classllvm_1_1DIModule.html

Regarding the spec compare [Literal Number Line](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/spirv-extensions/SPV_INTEL_debug_module.asciidoc)
Vs
[<id> Line](https://github.com/KhronosGroup/SPIRV-Registry/blob/1236f2b2fc1b1d32ef4a8afabfe6f7fe9e86f686/nonsemantic/NonSemantic.Shader.DebugInfo.200.asciidoc)